pallets/testnet/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, traits::Get,
+    weights::Weight,
+};
+use frame_system::{ensure_signed, RawOrigin};
+use pallet_identity::PermissionedCallOriginData;
+use polymesh_common_utilities::{
+    protocol_fee::ProtocolOp, traits::identity::Trait as IdentityTrait,
+};
+use polymesh_primitives::{
+    secondary_key::api::SecondaryKey, CddId, Claim, IdentityId, InvestorUid,
+};
+use sp_std::{prelude::*, vec};
+
+type Identity<T> = pallet_identity::Module<T>;
+type CallPermissions<T> = pallet_permissions::Module<T>;
+
+pub trait WeightInfo {
+    fn register_did(i: u32) -> Weight;
+    fn mock_cdd_register_did() -> Weight;
+}
+
+pub trait Trait: IdentityTrait {
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+    /// Weight information for extrinsics in the identity pallet.
+    type WeightInfo: WeightInfo;
+}
+
+decl_event!(
+    pub enum Event<T>
+    where
+        AccountId = <T as frame_system::Trait>::AccountId,
+    {
+        /// A new mocked `InvestorUid` has been created for the given Identity.
+        /// (Target DID, New InvestorUi)
+        MockInvestorUIDCreated(IdentityId, InvestorUid),
+        /// Emits the `IdentityId` and the `AccountId` of the caller.
+        /// (Caller DID, Caller account)
+        DidStatus(IdentityId, AccountId),
+        /// Shows the `DID` associated to the `AccountId`, and a flag indicates if that DID has a
+        /// valid CDD claim.
+        /// (Target DID, Target Account, a valid CDD claim exists)
+        CddStatus(Option<IdentityId>, AccountId, bool),
+    }
+);
+
+decl_storage! {
+    trait Store for Module<T: Trait> as testnet {
+    }
+}
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Register a new did with a CDD claim for the caller.
```

Comment:
Not really part of this task, but we could take the opportunity to document this extrinsic better now? or leave it be for later.. up to you :)